src/main/java/de/tum/in/www1/artemis/service/BuildLogEntryService.java

```diff
@@ -83,31 +82,57 @@ public boolean isIllegalReflectionLog(String logString) {
-        boolean isInfoWarningOrErrorLog = isInfoLog(logString) || isWarningLog(logString) || isErrorLog(logString) || isDockerImageLog(logString);
-        if (ProgrammingLanguage.JAVA.equals(programmingLanguage)) {
-            return isInfoWarningOrErrorLog || logString.startsWith("Unable to publish artifact") || logString.startsWith("NOTE: Picked up JDK_JAVA_OPTIONS")
-                    || logString.startsWith("Picked up JAVA_TOOL_OPTIONS") || logString.startsWith("[withMaven]") || logString.startsWith("$ docker");
+        boolean isUnnecessaryLog = isInfoLog(logString) || isWarningLog(logString) || isDockerImageLog(logString) || isGitLog(logString) || isTaskLog(logString);
+        if (isUnnecessaryLog) {
+            return true;
+        }
+        if (programmingLanguage == ProgrammingLanguage.JAVA) {
+            return isUnnecessaryJavaLog(logString);
         }
-        else if (ProgrammingLanguage.SWIFT.equals(programmingLanguage) || ProgrammingLanguage.C.equals(programmingLanguage)) {
-            return isInfoWarningOrErrorLog || logString.contains("Unable to find image") || logString.contains(": Already exists") || logString.contains(": Pull")
-                    || logString.contains(": Waiting") || logString.contains(": Verifying") || logString.contains(": Download") || logString.startsWith("Digest:")
-                    || logString.startsWith("Status:") || logString.contains("github.com");
+        else if (programmingLanguage == ProgrammingLanguage.SWIFT || programmingLanguage == ProgrammingLanguage.C) {
+            return isUnnecessarySwiftLog(logString);
         }
-        return isInfoWarningOrErrorLog;
+        return false;
+    }
+
+    private static final Set<String> UNNECESSARY_JAVA_LOGS = Set.of("NOTE: Picked up JDK_JAVA_OPTIONS", "Picked up JAVA_TOOL_OPTIONS", "[withMaven]", "$ docker");
+
+    private boolean isUnnecessaryJavaLog(String log) {
+        return isMavenErrorLog(log) || isGradleErrorLog(log) || isGradleInfoLog(log) || UNNECESSARY_JAVA_LOGS.stream().anyMatch(log::startsWith);
+    }
+
+    private static final Set<String> UNNECESSARY_SWIFT_LOGS = Set.of("Unable to find image", ": Already exists", ": Pull", ": Waiting", ": Verifying", ": Download");
+
+    private boolean isUnnecessarySwiftLog(String log) {
+        return UNNECESSARY_SWIFT_LOGS.stream().anyMatch(log::contains) || log.startsWith("Digest:") || log.startsWith("Status:") || log.contains("github.com");
     }
 
     private boolean isInfoLog(String log) {
-        return (log.startsWith("[INFO]") && !log.contains("error")) || log.startsWith("[INFO] Downloading") || log.startsWith("[INFO] Downloaded");
+        return (log.startsWith("[INFO]") && !log.contains("error")) || log.contains("Downloading") || log.contains("Downloaded") || log.contains("Progress (");
+    }
+
+    private static final Set<String> GRADLE_LOGS = Set.of("Downloading https://services.gradle.org", "...........10%", "Here are the highlights of this release:", "- ",
```

Comment:
Is "...........20%" not possible as log?

Reply:
No, this is always one line indicating the download progress. It contains a number every 10%, but I just added the start here.